browser/debounce/debounce_browsertest.cc

```diff
@@ -219,6 +219,16 @@ IN_PROC_BROWSER_TEST_F(DebounceBrowserTest, Redirect) {
   NavigateToURLAndWaitForRedirects(original_url, landing_url);
 }
 
+// Check that URLs ending with a '.' are properly debounced.
+IN_PROC_BROWSER_TEST_F(DebounceBrowserTest, URLThatEndsWithADot) {
```

Comment:
seems like some of these could be unit tests? wdyt?

Reply:
I don't see any unit tests for request_otr_rule, but debounce_rule has some. They are significantly faster to run if they can still adequately test the edge cases. We should always have browser tests for the more general cases, but unit tests are a great way to speed things up.

Reply:
if you feel like these need browser tests that's ok, too. Just checking to see if we can cut down on browser tests here.